axelor-business-project/src/main/java/com/axelor/apps/businessproject/service/ProductTaskTemplateServiceImpl.java

```diff
@@ -0,0 +1,73 @@
+package com.axelor.apps.businessproject.service;
+
+import com.axelor.apps.businessproject.db.ProductTaskTemplate;
+import com.axelor.apps.businessproject.db.repo.ProductTaskTemplateRepository;
+import com.axelor.apps.project.db.Project;
+import com.axelor.apps.project.db.TaskTemplate;
+import com.axelor.exception.AxelorException;
+import com.axelor.team.db.TeamTask;
+import com.axelor.team.db.repo.TeamTaskRepository;
+import com.google.inject.Inject;
+import com.google.inject.persist.Transactional;
+import java.math.BigDecimal;
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.List;
+
+public class ProductTaskTemplateServiceImpl implements ProductTaskTemplateService {
+
+  protected ProductTaskTemplateRepository repository;
+  protected TeamTaskBusinessService teamTaskBusinessService;
+  protected TeamTaskRepository teamTaskRepository;
+
+  @Inject
+  public ProductTaskTemplateServiceImpl(
+      ProductTaskTemplateRepository repository,
+      TeamTaskBusinessService teamTaskBusinessService,
+      TeamTaskRepository teamTaskRepository) {
+    this.repository = repository;
+    this.teamTaskBusinessService = teamTaskBusinessService;
+    this.teamTaskRepository = teamTaskRepository;
+  }
+
+  @Override
+  @Transactional(rollbackOn = {AxelorException.class, Exception.class})
```

Comment:
It works, but it makes more sense here to write `@Transactional`, as no axelor exception is thrown. See https://github.com/google/guice/wiki/Transactions#responding-to-exceptions